rclpy/src/rclpy/logging.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rcl/error_handling.h>
+#include <rcl/logging.h>
+#include <rcl/rcl.h>
+#include <rcl/types.h>
+#include <rcutils/logging.h>
+
+#include <mutex>
+#include <stdexcept>
+
+#include "rclpy_common/exceptions.hpp"
+#include "rclpy_common/handle.h"
+
+#include "logging.hpp"
+
+extern "C"
+{
```

Comment:
@sloretz meta: why the `extern "C"` block? AFAIK it only affects the function symbol, C and C++ are ABI compatible.

Reply:
On a related note, we should probably ensure symbol locality (i.e. `static`, anonymous namespace).

Reply:
Oops, I mistakenly thought extern "C" changed the calling convention, but it does not. Removed `extern C` stuff and made appropriate functions `static` in 570b63d7d1751b61099fd705fb22133e05ccdf38